From 35098a8f8be2a6a715e213faa48e1a6b0bf5a1f4 Mon Sep 17 00:00:00 2001
Message-Id: <35098a8f8be2a6a715e213faa48e1a6b0bf5a1f4.1429601425.git.chang-joon.lee@intel.com>
In-Reply-To: <20df33cbe9e95d11653023b1a6201653c517d353.1429601425.git.chang-joon.lee@intel.com>
References: <20df33cbe9e95d11653023b1a6201653c517d353.1429601425.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Fri, 17 Apr 2015 18:54:06 +0530
Subject: [PATCH 9/9] SQUASHME! [VPG]: drm/i915: update the wm, if ddr dvfs
 feature is supported.

There is no need to update the intermediate watermarks if we are
not supporting ddr dvfs.

SQUASHME: This patch needs to be squashed with
	commit 652ff6144d7eddfc80598426604634ab52e3a6da
	Author: vandita kulkarni <vandita.kulkarni@intel.com>
	Date:   Wed Mar 25 20:04:03 2015 +0530

		SQUASHME!: [VPG]: drm/i915: restructure maxfifo and add ddr dvfs.

Issue: GMINL-4468
Change-Id: I2ff627acad6a97e5a6f5aed2e8d5c4ef0fa718a4
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 16f3b66..db24d6f 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11119,8 +11119,9 @@ static int intel_crtc_set_display(struct drm_crtc *crtc,
 	prev_plane_stat = VLV_PLANE_STATS(dev_priv->prev_pipe_plane_stat, pipe);
 	plane_stat = VLV_PLANE_STATS(dev_priv->pipe_plane_stat, pipe);
 
-	if (hweight32(prev_plane_stat) <=  hweight32(plane_stat))
-		valleyview_update_wm_pm5(intel_crtc);
+	if (IS_CHERRYVIEW(dev))
+		if (hweight32(prev_plane_stat) <=  hweight32(plane_stat))
+			valleyview_update_wm_pm5(intel_crtc);
 
 	/* Check if we need to a vblank, if so wait for vblank */
 	if (intel_dsi_is_enc_on_crtc_cmd_mode(crtc)) {
-- 
1.7.9.5

